src/main/java/nvt/kts/ticketapp/service/event/EventServiceImpl.java

```diff
@@ -0,0 +1,148 @@
+
+        Event event = new Event(eventEventDaysDTO.getEvent().getName(), eventEventDaysDTO.getEvent().getCategory(), eventEventDaysDTO.getEvent().getDescription());
+
+        List<EventDay> eventDays = new ArrayList<>();
+        List<Location> locations = new ArrayList<>();
+        List<LocationSector> locationSectors = new ArrayList<>();
+
+        if (eventEventDaysDTO.getEventDays().isEmpty()) {
+            throw new EventDaysListEmpty();
+        }
+
+        // create eventDays
+        for (EventDayDTO eventDayDTO: eventEventDaysDTO.getEventDays()) {
+
+            Date date = parseDate(eventDayDTO.getDate());
+            Date reservationExpireDate = parseDate(eventDayDTO.getReservationExpireDate());
+
+            // check if dates in past or reservationExpireDate is before event date
+            checkDates(date, reservationExpireDate);
+
+            Long locationSchemeId = eventDayDTO.getLocation().getLocationSchemeId();
+            LocationScheme locationScheme = locationSchemeService.findOneById(locationSchemeId);
+
+            // check if a location is available that date
+            checkAvailabilityOfLocationOnDate(locationScheme.getId(), date);
+
+            Location location = new Location(locationScheme);
+            locations.add(location);
+
+            EventDay eventDay = new EventDay(date, location, reservationExpireDate, EventDayState.NOT_IN_SALE, event);
+            eventDays.add(eventDay);
+
+            // create locationSectors
+            for (LocationSectorsDTO locationSectorsDTO : eventDayDTO.getLocation().getLocationSectors()) {
+
+                Sector sector = sectorService.findOneById(locationSectorsDTO.getSectorId());
+
+                if (sector.getCapacity() < locationSectorsDTO.getCapacity()) {
+                    throw new SectorCapacityOverload();
+                }
+                LocationSector locationSector = new LocationSector(sector, location, locationSectorsDTO.getPrice(), locationSectorsDTO.getCapacity(), locationSectorsDTO.isVip());
+                locationSectors.add(locationSector);
+            }
+
+        }
+
+        // save event, eventDays, locations, locationSectors
+        event = eventRepository.save(event);
+        locationService.saveAll(locations);
+        locationSectorService.saveAll(locationSectors);
+        eventDayService.saveAll(eventDays);
+
+        return event;
+    }
+
+    private void checkDates(Date date, Date reservationExpireDate) throws DateCantBeInPast, ReservationExpireDateInvalid {
+
+        if (dateInPast(date) || dateInPast(reservationExpireDate)) {
+            throw new DateCantBeInPast();
+        }
+
+        if (reservationExpireDate.after(date)) {
+            throw new ReservationExpireDateInvalid();
+        }
+
+    }
+
+    private void checkAvailabilityOfLocationOnDate(Long locationSchemeId, Date date) throws ParseException, LocationNotAvailableThatDate {
+
+        List<EventDay> eventDays = eventDayService.getAll();
```

Comment:
Mislim da mozda moze da se napravi query koji bi izgledao:
findAllByDate i da se prosledi datum u ubliku u kom se cuva, pa ce ti repozitorijum vratiti sve sa tim datumom.

Cak,
findAllByDateAndLocationName(date, name)

skratice kod a imas to kod profe na slajdovima kako se grade ti upiti

Reply:
Moglo bi da se dobave eventDay koji su tog datuma, to ce mi samo smanjiti proveru da li su datumi isti, svakako cu morati prolaziti kroz listu.